phoenix-core/src/main/java/org/apache/phoenix/schema/types/PDataType.java

```diff
@@ -94,6 +94,21 @@ public boolean isBytesComparableWith(PDataType otherType) {
         return equalsAny(this, otherType, PVarbinary.INSTANCE, PBinary.INSTANCE);
     }
 
+    /**
+     * @return true if {@link PDataType} can be declared as primary key otherwise false.
+     */
+    public boolean canBePrimaryKey() {
+        return true;
+    }
+
+    /**
+     * @return true if {@link PDataType} supports equality operators (=,!=,<,>,<=,>=) otherwise
+     *         false.
+     */
+    public boolean isEqualitySupported() {
```

Comment:
Maybe rename this function to `isComparisonSupported`

Reply:
Done